ladders.js

```diff
@@ -274,11 +274,16 @@ class Ladder extends LadderStore {
 	 */
 	static removeChallenge(challenge, skipUpdate = false) {
 		const fromChalls = /** @type {Challenge[]} */ (challenges.get(challenge.from));
-		fromChalls.splice(fromChalls.indexOf(challenge), 1);
-		if (!fromChalls.length) challenges.delete(challenge.from);
+		if (fromChalls) {
+			fromChalls.splice(fromChalls.indexOf(challenge), 1);
+			if (!fromChalls.length) challenges.delete(challenge.from);
+		}
 		const toChalls = /** @type {Challenge[]} */ (challenges.get(challenge.to));
-		toChalls.splice(toChalls.indexOf(challenge), 1);
-		if (!toChalls.length) challenges.delete(challenge.to);
+		if (toChalls) {
```

Comment:
The way I fixed this was not to test `toChalls` at all, and instead early exit if `fromChalls` doesn't have the challenge.

Recently, I've been placing value on not suppressing inconsistent state crashes. In this case: The challenge still existing in `fromChalls` means the challenge should also still exist in `toChalls`.